lib/state-processor/state-processor.js

```diff
@@ -9,12 +9,14 @@ const Events = require('../constants/events');
 const errorUtils = require('../errors/utils');
 
 module.exports = class StateProcessor {
-    constructor(captureProcessorType) {
+    constructor(config, captureProcessorType) {
```

Comment:
- `captureProcessorType` is the most significant parameter, so it should be the first paramter
- why do you pass whole `config` object here? It would be nice to pass as minimum config section as needed